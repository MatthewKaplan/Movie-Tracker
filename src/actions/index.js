export const fetchPopularMovies = popularMoviess => ({
  type: "FETCH_POPULAR_MOVIES",
  favorited: false,
  popularMoviess
});

export const fetchPopularTv = popularShows => ({
  type: "FETCH_POPULAR_TV",
  favorited: false,
  popularShows
});

export const fetchUpcoming = upcomingResults => ({
  type: "FETCH_UPCOMING",
  favorited: false,
  upcomingResults
});

export const fetchNews = newsResults => ({
  type: "FETCH_NEWS",
  newsResults
});

export const fetchGenres = genreResults => ({
  type: "FETCH_GENRES",
  favorited: false,
  genreResults
});

export const fetchSearch = searchResults => ({
  type: "FETCH_SEARCH",
  favorited: false,
  searchResults
});

export const setUser = user => ({
  type: "SET_USER",
  user
});

export const createUser = user => ({
  type: "CREATE_USER",
  user
});

export const isLoggedIn = bool => ({
  type: "IS_LOGGED_IN",
  isLoggedIn: bool
});

export const favoritesList = movies => ({
  type: "FAVORITES_LIST",
  favorited: false,
  movies
});

export const currentMovie = movie => ({
  type: "CURRENT_MOVIE",
  movie
});

export const handleError = errorMessage => ({
  type: "HANDLE_ERROR",
  errorMessage
});
