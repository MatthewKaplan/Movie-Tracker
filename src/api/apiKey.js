export const apiKey = "api_key=bc13bb74ba2ec1d960ae15ea70d535e5";

export default apiKey;