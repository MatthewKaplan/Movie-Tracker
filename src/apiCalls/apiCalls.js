import { BASE_URL } from '../util/path';

const fetchData = url => {
	return fetch(url).then(response => {
		if (!response.ok) {
			throw new Error('Fetch failed');
		} else {
			return response.json();
		}
	});
};

const fetchUserData = url => {
	return fetch(url).then(response => {
		if (!response.ok) {
			throw new Error('Fetch failed');
		} else {
			return response.json();
		}
	});
};

const fetchPost = async (url, options) => {
	try {
		const response = await fetch(url, options);
		return response.json();
	} catch (e) {
		console.log('Unable to Add a new note', e);
	}
};

const fetchAddUser = async (name, email, password) => {
  const url = `${BASE_URL}/users`;
	const body = { email, name, password };
	const options = {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(body)
  };
  
	try {
		const response = await fetch(url, options);
		return response.json();
	} catch (e) {
		console.log('Unable to Add a new note', e);
	}
};

const fetchUserLogin = async (email, password) => {
  const url = `${BASE_URL}/users/login`;
  const body = {email, password};
  const options = {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(body)
  };
  
  try {
		const response = await fetch(url, options);
		return response.json();
	} catch (e) {
		console.log('Unable to Add a new note', e);
	}
};

const fetchFavoriteMovies = async (id) => {
  const url = `${BASE_URL}/movies/favorites/${id}`;

  try {
    const favorites = await fetch(url);
    return favorites.json();
  } catch (e) {
    console.log('Unable to Add a new note', e);
  }
}

export { fetchPost, fetchData, fetchUserData, fetchAddUser, fetchUserLogin, fetchFavoriteMovies };
