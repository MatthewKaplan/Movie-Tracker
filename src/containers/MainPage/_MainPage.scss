@import "../../variables.scss";

.main-page {
  margin: 0 auto;
  display: flex;
  position: relative;
  z-index: 2;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  background-color: $background-color;
  max-width: 1280px;
}

.under-splash {
  transform: translateY(-3%);
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  max-width: 1280px;
}

.container-title {
  color: $text-w;
  font-size: $sub-title;
  margin-left: 5px;
}

.container-titles {
  display: flex;
  justify-content: space-around;
}

.major-release-section {
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 100%;

  .main-release {
    @include hover-state(8.5);
    background-image: url("https://www.hypable.com/wp-content/uploads/2018/12/bladerunner2049ff.jpg");
    background-size: cover;
    height: 525px;
    width: 350px;
    &:nth-child(2) {
      margin: 0 2.5rem 0 0.8rem;
    }
  }
}

.genres {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  height: 575px;
  color: white;

  .genreTitle {
    font-size: 40px;
    margin-top: 0;
    -webkit-text-stroke-width: 1px;
    -webkit-text-stroke-color: black;
  }

  .horror {
    background-image: url("https://i.imgur.com/COmAJiV.jpg");
  }

  .documentaries {
    background-image: url("https://i.imgur.com/X2IBLKh.jpg");
    background-position: 50% 35%;
  }

  .action {
    background-image: url("https://i.imgur.com/FLII2dO.jpg");
  }

  .romance {
    background-image: url("https://i.imgur.com/S7CyGk2.jpg");
    background-position: 25% 25%;
  }

  .family {
    background-image: url("https://i.imgur.com/6JYnIoV.jpg");
    background-position: 50% 20%;
  }

  .comedy {
    background-image: url("https://i.imgur.com/4FY035F.jpg");
    background-position: 50% 30%;
  }
}
.genre {
  @include hover-state(8.5);
  background: url("https://assets.fontsinuse.com/static/use-media-items/50/49197/full-2700x3910/58b6d005/Moonlight_poster.jpeg?resolution=0");
  background-size: cover;
  background-position: center;
  height: 175px;
  width: 48%;
  margin: 0.4rem;
}

.carousel-section,
.major-release-container,
.genre-container {
  margin-top: 2.5rem;
  // border: 1px solid red;
  width: 100%;
  h1 {
    // border: 1px solid red;
    padding: 1rem 0 0 0.7rem;
  }
}

.searchCardResults {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 80px;
  margin-top: 200px;
}
