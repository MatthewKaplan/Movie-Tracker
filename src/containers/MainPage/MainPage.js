import React from 'react';
import SplashImage from '../SplashImage/SplashImage';
import Carousel from '../Carousel/Carousel';
import MovieCard from '../MovieCard/MovieCard';
import { NavLink, Link } from 'react-router-dom';
import GenrePage from '../GenrePage/GenrePage';
import News from '../News/News';
import { apiKey } from '../../api/apiKey';
import { nytApiKey } from '../../api/nytApiKey';
import Footer from '../Footer/Footer';
import PropTypes from 'prop-types';
import './_MainPage.scss';
import * as actions from '../../actions/index';
import { connect } from 'react-redux';
import { fetchData } from '../../apiCalls/apiCalls';
import MajorReleases from '../MajorReleases/MajorReleases';

export class MainPage extends React.Component {
	state = {
		movieUrl: 'https://api.themoviedb.org/3',
		newsUrl: 'https://api.nytimes.com/svc',
		showPopUp: false,
		error: ''
	};
	componentDidMount () {
		this.fetchNewsResults();
		this.fetchPopularMovies();
		this.fetchPopularTvShows();
		this.fetchComingSoon();
	}

	fetchPopularMovies = async () => {
		const url = `${this.state.movieUrl}/discover/movie?${apiKey}&sort_by=popularity.desc`;
		try {
			const popularMovies = await fetchData(url);
			this.props.fetchPopularMovies(popularMovies.results);
		} catch (error) {
			this.setState({ error: error.message });
		}
	};

	renderPopularMovies = () => {
		return this.props.movies.map(movie => (
			<MovieCard key={movie.id} wholeObj={movie} title={movie.title} img={movie.backdrop_path} />
		));
	};

	fetchPopularTvShows = async () => {
		const url = `${this.state.movieUrl}/discover/tv?${apiKey}&sort_by=popularity.desc`;
		try {
			const popularTv = await fetchData(url);
			this.props.fetchPopularTv(popularTv.results);
		} catch (error) {
			this.setState({ error: error.message });
		}
	};

	renderPopularTvShows = () => {
		return this.props.tv.map(tvShow => (
			<MovieCard key={tvShow.id} wholeObj={tvShow} title={tvShow.name} img={tvShow.backdrop_path} />
		));
	};

	fetchComingSoon = async () => {
		const url = `${this.state.movieUrl}/movie/upcoming?${apiKey}`;
		try {
			const upcoming = await fetchData(url);
			this.props.fetchUpcoming(upcoming.results);
		} catch (error) {
			this.setState({ error: error.message });
		}
	};

	renderComingSoon = () => {
		return this.props.upcoming.map(film => (
			<MovieCard key={film.id} wholeObj={film} title={film.title} img={film.backdrop_path} />
		));
	};

	renderFeaturedMovies = () => {
		let filteredMovies = [];
		for (let i = 0; i < 2; i++) {
			filteredMovies.push(this.props.movies[i]);
		}
		if (this.props.movies.length > 0) {
			return filteredMovies.map(movie => <MajorReleases key={movie.id} wholeObj={movie} img={movie.poster_path} />);
		}
	};

	fetchNewsResults = async () => {
		const url = `${this.state.newsUrl}/topstories/v2/movies.json?${nytApiKey}`;
		try {
			const newsResults = await fetchData(url);
			this.props.fetchNews(newsResults.results);
		} catch (error) {
			this.setState({ error: error.message });
		}
	};

	fetchGenre = async endPath => {
		const url = `${this.state.movieUrl}/discover/movie?${apiKey}${endPath}&page=3`;
		try {
			const genreResults = await fetchData(url);
			this.props.fetchGenres(genreResults.results);
		} catch (error) {
			this.setState({ error: error.message });
		}
	};

	render () {
		return (
			<main className="main-page">
				<div>
					<div className="splash-movie">
						<SplashImage />
					</div>
					<div className="under-splash">
						<section className="major-release-container">
							<div className="container-titles">
								<h1 className="container-title">Major Releases</h1>
								<h1 className="container-title">Movie News</h1>
							</div>
							<section className="major-release-section">
								{this.renderFeaturedMovies()}
								<div className="movie-news">
									<News />
								</div>
							</section>
						</section>
						<section className="genre-container">
							<h1 className="container-title">Genres</h1>
							<section className="genres">
								<NavLink
									to="/ActionMovies"
									data-test="action-movies-btn"
									className="genre action"
									onClick={() => this.fetchGenre('&with_genres=28')}>
									<div>
										<h1 className="genreTitle">Action</h1>
									</div>
								</NavLink>
								<Link
									to="/ComedyMovies"
									data-test="comedy-movies-btn"
									className="genre comedy"
									onClick={() => this.fetchGenre('&with_genres=35')}>
									<div>
										<h1 className="genreTitle">Comedy</h1>
									</div>
								</Link>
								<Link
									to="/Documentaries"
									data-test="documentaries-btn"
									className="genre documentaries"
									onClick={() => this.fetchGenre('&with_genres=99')}>
									<div>
										<h1 className="genreTitle">Documentaries</h1>
									</div>
								</Link>
								<Link
									to="/FamilyMovies"
									data-test="family-movies-btn"
									className="genre family"
									onClick={() => this.fetchGenre('&with_genres=10751')}>
									<div>
										<h1 className="genreTitle">Family</h1>
									</div>
								</Link>
								<Link
									to="/HorrorMovies"
									data-test="horror-movies-btn"
									className="genre horror"
									onClick={() => this.fetchGenre('&with_genres=27')}>
									<div>
										<h1 className="genreTitle">Horror</h1>
									</div>
								</Link>
								<Link
									to="/RomanceMovies"
									data-test="romance-movies-btn"
									component={GenrePage}
									className="genre romance"
									onClick={() => this.fetchGenre('&with_genres=10749')}>
									<div>
										<h1 className="genreTitle">Romance</h1>
									</div>
								</Link>
							</section>
							<section className="carousel-section">
								<h1>Popular Movies</h1>
								<Carousel>{this.renderPopularMovies()}</Carousel>
								<h1>Popular TV Series</h1>
								<Carousel>{this.renderPopularTvShows()}</Carousel>
								<h1>Coming to theaters soon</h1>
								<Carousel>{this.renderComingSoon()}</Carousel>
							</section>
						</section>
					</div>
				</div>
				<div className="page-footer">
					<Footer />
				</div>
			</main>
		);
	}
}

MainPage.propTypes = {
	movies: PropTypes.array.isRequired,
	tv: PropTypes.array.isRequired,
	upcoming: PropTypes.array.isRequired,
	news: PropTypes.array,
	genre: PropTypes.array.isRequired,
	fetchPopularMovies: PropTypes.func,
	fetchPopularTv: PropTypes.func,
	fetchUpcoming: PropTypes.func,
	fetchNews: PropTypes.func,
	fetchGenres: PropTypes.func
};

export const mapStateToProps = state => {
	return {
		movies: state.movies,
		tv: state.tv,
		upcoming: state.upcoming,
		news: state.news,
		genre: state.genre
	};
};

export const mapDispatchToProps = dispatch => ({
	fetchPopularMovies: popularMovies => dispatch(actions.fetchPopularMovies(popularMovies)),
	fetchPopularTv: popularTv => dispatch(actions.fetchPopularTv(popularTv)),
	fetchUpcoming: upcoming => dispatch(actions.fetchUpcoming(upcoming)),
	fetchNews: news => dispatch(actions.fetchNews(news)),
	fetchGenres: genres => dispatch(actions.fetchGenres(genres))
});

export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
