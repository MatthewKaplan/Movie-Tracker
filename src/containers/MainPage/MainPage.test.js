import React from "react";
import { shallow } from "enzyme";
import { MainPage, mapStateToProps, mapDispatchToProps } from "./MainPage";
import * as actions from "../../actions/index";
import MockData from "../../assets/mockData";
import { fetchData } from "../../apiCalls/apiCalls";
import { nytApiKey } from "../../api/nytApiKey";
import { apiKey } from "../../api/apiKey";

jest.mock("../../apiCalls/apiCalls.js");

let mockMoviesProps = MockData.popularMovies;
let mockTvProps = MockData.popularShows;
let mockFavoritesProp = MockData.favoritedMovies;
let mockSeachProps = MockData.searchResults;
let mockGenreProps = MockData.genreResults;
let mockUpcoming = MockData.upcomingMovies;
let mockNewsResults = MockData.mockNewsResults;
let mockLocation = jest.fn();
let mockFilteredMovies = MockData.filteredMovie;
const mockFetchPopularMovies = jest.fn();
const mockFetchUpcoming = jest.fn();
const mockFetchPopularTv = jest.fn();
const mockFetchNews = jest.fn();
const mockFetchGenres = jest.fn();

describe("MainPage", () => {
  let wrapper, instance;

  fetchData.mockImplementation(() => Promise.resolve(1));

  beforeEach(() => {
    wrapper = shallow(
      <MainPage
        location={mockLocation}
        movies={mockMoviesProps}
        tv={mockTvProps}
        favoriteList={mockFavoritesProp}
        searchResults={mockSeachProps}
        genre={mockGenreProps}
        upcoming={mockUpcoming}
        filteredMovies={mockFilteredMovies}
        fetchPopularMovies={mockFetchPopularMovies}
        fetchUpcoming={mockFetchUpcoming}
        fetchPopularTv={mockFetchPopularTv}
        fetchNews={mockFetchNews}
        fetchGenres={mockFetchGenres}
      />
    );
    instance = wrapper.instance();
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        ok: true,
        json: () => Promise.resolve(1)
      })
    );
  });

  afterEach(() => {
    fetchData.mockClear();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'fetchNewsResults, fetchPopularMovies, fetchPopularTvShows, fetchComingSoon' when mounted", () => {
    jest.spyOn(instance, "fetchNewsResults");
    jest.spyOn(instance, "fetchPopularMovies");
    jest.spyOn(instance, "fetchPopularTvShows");
    jest.spyOn(instance, "fetchComingSoon");
    instance.componentDidMount();
    expect(instance.fetchNewsResults).toHaveBeenCalled();
    expect(instance.fetchPopularMovies).toHaveBeenCalled();
    expect(instance.fetchPopularTvShows).toHaveBeenCalled();
    expect(instance.fetchComingSoon).toHaveBeenCalled();
  });

  describe("fetchPopularMovies", () => {
    it("should call 'fetchData' with correct params when 'fetchPopularMovies' is invoked", () => {
      instance.fetchPopularMovies();
      expect(fetchData).toHaveBeenCalledWith(
        wrapper.state("movieUrl") +
          "/discover/movie?" +
          apiKey +
          "&sort_by=popularity.desc"
      );
    });

    it("should invoke 'fetchPopularMovies' with the results from 'fetchData'", () => {
      instance.fetchPopularMovies();
      expect(mockFetchPopularMovies).toHaveBeenCalled();
    });

    it("should finally throw an error if the response is not ok and save it to local state", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await instance.fetchPopularMovies();
      expect(wrapper.state("error")).toBe("Fetch failed");
    });
  });

  describe("fetchPopularTvShows", () => {
    it("should call 'fetchData' with correct params when 'fetchPopularTvShows' is invoked", () => {
      instance.fetchPopularTvShows();
      expect(fetchData).toHaveBeenCalledWith(
        wrapper.state("movieUrl") +
          "/discover/tv?" +
          apiKey +
          "&sort_by=popularity.desc"
      );
    });

    it("should invoke 'fetchPopularTv' with the results from 'fetchData'", () => {
      instance.fetchPopularTvShows();
      expect(mockFetchPopularTv).toHaveBeenCalled();
    });

    it("should finally throw an error if the response is not ok and save it to local state", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await instance.fetchPopularTvShows();
      expect(wrapper.state("error")).toBe("Fetch failed");
    });
  });

  describe("fetchComingSoon", () => {
    it("should call 'fetchData' with correct params when 'fetchComingSoon' is invoked", () => {
      instance.fetchComingSoon();
      expect(fetchData).toHaveBeenCalledWith(
        wrapper.state("movieUrl") + "/movie/upcoming?" + apiKey
      );
    });

    it("should invoke 'fetchUpcoming' with the results from 'fetchData'", () => {
      instance.fetchComingSoon();
      expect(mockFetchUpcoming).toHaveBeenCalled();
    });

    it("should finally throw an error if the response is not ok and save it to local state", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await instance.fetchComingSoon();
      expect(wrapper.state("error")).toBe("Fetch failed");
    });
  });

  describe("fetchComingSoon", () => {
    it("should call 'fetchData' with correct params when 'fetchNewsResults' is invoked", () => {
      instance.fetchNewsResults();
      expect(fetchData).toHaveBeenCalledWith(
        wrapper.state("newsUrl") + "/topstories/v2/movies.json?" + nytApiKey
      );
    });

    it("should invoke 'fetchNews' with the results from 'fetchData'", () => {
      instance.fetchNewsResults();
      expect(mockFetchNews).toHaveBeenCalled();
    });

    it("should finally throw an error if the response is not ok and save it to local state", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await instance.fetchNewsResults();
      expect(wrapper.state("error")).toBe("Fetch failed");
    });
  });

  describe("fetchGenre", () => {
    it("should call 'fetchData' with correct params when 'fetchGenre' is invoked", () => {
      const path = "&with_genres=10749";
      instance.fetchGenre(path);
      expect(fetchData).toHaveBeenCalledWith(
        wrapper.state("movieUrl") + "/discover/movie?" + apiKey + path + "&page=3"
      );
    });

    it("should invoke 'fetchGenres' with the results from 'fetchData'", () => {
      instance.fetchGenre();
      expect(mockFetchGenres).toHaveBeenCalled();
    });

    it("should finally throw an error if the response is not ok and save it to local state", async () => {
      fetchData.mockImplementationOnce(() =>
        Promise.reject(new Error("Fetch failed"))
      );
      await instance.fetchGenre();
      expect(wrapper.state("error")).toBe("Fetch failed");
    });

    it("should invoke 'fetchGenre' to fetch action movies when action button is clicked", () => {
      const path = "&with_genres=28";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='action-movies-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });

    it("should invoke 'fetchGenre' to fetch comedy movies when comedy button is clicked", () => {
      const path = "&with_genres=35";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='comedy-movies-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });

    it("should invoke 'fetchGenre' to fetch documentaries when documentaries button is clicked", () => {
      const path = "&with_genres=99";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='documentaries-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });

    it("should invoke 'fetchGenre' to fetch family movies when family button is clicked", () => {
      const path = "&with_genres=10751";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='family-movies-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });

    it("should invoke 'fetchGenre' to fetch horror movies when horror button is clicked", () => {
      const path = "&with_genres=27";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='horror-movies-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });

    it("should invoke 'fetchGenre' to fetch romance movies when romance button is clicked", () => {
      const path = "&with_genres=10749";
      jest.spyOn(instance, "fetchGenre");
      wrapper.find("[data-test='romance-movies-btn']").simulate("click");
      expect(instance.fetchGenre).toHaveBeenCalledWith(path);
    });
  })
});

describe("mapStateToProps", () => {
  it("should return an object", () => {
    const mockData = {
      movies: [],
      tv: [],
      upcoming: [],
      news: [],
      genre: []
    };
    const expected = {
      movies: [],
      tv: [],
      upcoming: [],
      news: [],
      genre: []
    };

    const mockprops = mapStateToProps(mockData);
    expect(mockprops).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  it("should call dispatch for fetchPopularMovies", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.fetchPopularMovies(mockMoviesProps);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchPopularMovies(mockMoviesProps);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for fetchPopularTv", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.fetchPopularTv(mockTvProps);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchPopularTv(mockTvProps);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for fetchUpcoming", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.fetchUpcoming(mockUpcoming);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchUpcoming(mockUpcoming);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for fetchNews", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.fetchNews(mockNewsResults);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchNews(mockNewsResults);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should call dispatch for fetchGenres", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.fetchGenres(mockGenreProps);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchGenres(mockGenreProps);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
