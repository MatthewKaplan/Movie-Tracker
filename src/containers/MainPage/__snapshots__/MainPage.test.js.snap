// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renderFeaturedMovies should match snapshot if movies length is less than zero 1`] = `
<MainPage
  movies={Array []}
/>
`;

exports[`MainPage should match the snapshot 1`] = `ShallowWrapper {}`;
